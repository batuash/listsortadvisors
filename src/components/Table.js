import React from "react";
import { compose } from "lodash/fp";
import { withSort, withFilter, withInfiniteScroll } from "../hoc/";

const TABLE_COLUMNS = [
  { id: "fname", title: "First Name" },
  { id: "lname", title: "Last Name" },
  { id: "status", title: "Status" },
  { id: "language", title: "Language" },
  { id: "reviews", title: "Reviews" }
];

const Table = props => {
  const { applySort, sortRows, applyFilter, filterRows, advisors } = props;

  return (
    <table>
      <thead>
        <tr>
          {TABLE_COLUMNS.map(({ id, title }, idx) => (
            <th key={`${title}_${idx}`} onClick={applySort(id)}>
              {id === "number" ? "#" : title}
            </th>
          ))}
        </tr>
        <tr className="Table-filter-row">
          {TABLE_COLUMNS.map(({ id, title }, idx) => (
            <th key={`filter_${title}_${idx}`}>
              <input
                type="text"
                onChange={applyFilter(id)}
                placeholder="Filter"
                className="Table-filter-input"
              />
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {compose(
          sortRows,
          filterRows
        )(advisors).map((row, rowIdx) => (
          <tr key={`${row.number}_${rowIdx}`} className="Table-body-row">
            {TABLE_COLUMNS.map(({ id }, celIdx) => (
              <td key={`${row.number}_${celIdx}`}>{row[id]}</td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

Table.defaultProps = {
  advisors: []
};

export default compose(
  withInfiniteScroll,
  withSort,
  withFilter
)(Table);
