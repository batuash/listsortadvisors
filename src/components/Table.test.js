import React from 'react';
import Table from './Table';
import ShallowRenderer from 'react-test-renderer/shallow';
import tableData from '../static/sampleData'


describe('Table', () => {
	it('renders correctly with no data', () => {
		const renderer = new ShallowRenderer();
	  const tree = renderer.render(<Table />);
	  expect(tree).toMatchSnapshot();
	});

	it('renders correctly with valid data', () => {
		const renderer = new ShallowRenderer();
	  const tree = renderer.render(<Table tableData={tableData} />);
	  expect(tree).toMatchSnapshot();
	});
});
