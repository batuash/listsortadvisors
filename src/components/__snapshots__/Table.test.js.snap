// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly with no data 1`] = `
<_temp
  applySort={[Function]}
  desc={false}
  orderBy="number"
  sortRows={[Function]}
/>
`;

exports[`Table renders correctly with valid data 1`] = `
<_temp
  applySort={[Function]}
  desc={false}
  orderBy="number"
  sortRows={[Function]}
  tableData={
    Object {
      "columns": Array [
        Object {
          "id": "number",
          "title": "Number",
        },
        Object {
          "id": "fname",
          "title": "First Name",
        },
        Object {
          "id": "lname",
          "title": "Last Name",
        },
        Object {
          "id": "status",
          "title": "Status",
        },
        Object {
          "id": "language",
          "title": "Language",
        },
        Object {
          "id": "reviews",
          "title": "Reviews",
        },
      ],
      "rows": Array [
        Object {
          "fname": "The",
          "language": "es",
          "lname": "Rock",
          "number": 1,
          "reviews": 100,
          "status": "online",
        },
        Object {
          "fname": "Stone",
          "language": "en",
          "lname": "Cold",
          "number": 2,
          "reviews": 143,
          "status": "online",
        },
        Object {
          "fname": "The",
          "language": "de",
          "lname": "Undertaker",
          "number": 3,
          "reviews": 67,
          "status": "offline",
        },
      ],
    }
  }
/>
`;
